key_config:
  # Hardcoded : link to the code
  topic_types:
    'switch_teleop':
      # Pour rappel
      'usage': 'No value'
      'structure': ['type']
      # Utile pour le code
      'import': ['std_msgs.msg', 'String']

    'twist':
      # Pour rappel
      'usage': ['lin_x','lin_y','ang_z']
      'structure': ['type', 'range_lin', 'range_ang', 'precision']
      # Utile pour le code
      'import': ['geometry_msgs.msg', 'Twist']
    'wrench':
      # Pour rappel
      'usage': ['lin_x','lin_y','ang_z']
      'structure': ['type', 'range_lin', 'range_ang', 'precision']
      # Utile pour le code
      'import': ['geometry_msgs.msg', 'Wrench']
    'position':
      # Pour rappel
      'usage': 'position'
      'structure': ['type', 'range', 'precision']
      # Utile pour le code
      'import': ['std_msgs.msg', 'Float32']
    'int':
      # Pour rappel
      'usage': ['value']
      'structure': ['type']
      # Utile pour le code
      'import': ['std_msgs.msg', 'Int8']
    'string':
      # Pour rappel
      'usage': ['value']
      'structure': ['type']
      # Utile pour le code
      'import': ['std_msgs.msg', 'String']

  # Modifiable
  topic_map:
    'switch_teleop':
      'type': 'switch_teleop'

    'twist':
      'type': 'twist'
      'range_lin': 1.0
      'range_ang': 1.0
      'precision': 0.2

    'state':
      'type': 'int'

  key_map:
    # switch teleop key
    't':
      topic: 'switch_teleop'

    # Twist keys
    ' ':
      topic: 'twist'
      value: ['STOP','STOP','STOP']
    'r':
      topic: 'twist'
      value: ['BACK','BACK','BACK']
    'z':
      topic: 'twist'
      value: [1,0,0]
    's':
      topic: 'twist'
      value: [-1,0,0]
    'q':
      topic: 'twist'
      value: [0,0,1]
    'd':
      topic: 'twist'
      value: [0,0,-1]
    'a':
      topic: 'twist'
      value: [0,0,1]
    'e':
      topic: 'twist'
      value: [0,0,-1]

    # Cmd state keys
    'm':
      topic: 'state'
      value: 0
    'l':
      topic: 'state'
      value: 1
